src/platform.deno.ts

```diff
@@ -64,17 +74,45 @@ export class InputFile {
     constructor(
         file:
             | string
+            | URL
+            | URLLike
             | Uint8Array
             | ReadableStream<Uint8Array>
             | AsyncIterable<Uint8Array>,
         filename?: string,
     ) {
-        this[inputFileData] = file;
+        this.fileData = file;
         if (filename === undefined && typeof file === "string") {
             filename = basename(file);
         }
         this.filename = filename;
     }
+    get [inputFileData]() {
+        if (this.consumed) {
+            throw new Error("Cannot reuse InputFile data source!");
+        }
+        let data = this.fileData;
+        if (
+            typeof data === "object" && ("url" in data || data instanceof URL)
+        ) {
+            data = fetchFile(data instanceof URL ? data : data.url);
+        } else if (
+            typeof data !== "string" && (!(data instanceof Uint8Array))
+        ) {
+            this.consumed = false;
+        }
+        return data;
+    }
+}
+
+async function* fetchFile(url: string | URL): AsyncIterable<Uint8Array> {
+    const { body } = await fetch(url);
+    if (body === null) {
+        throw new Error(
+            `Download failed, no response body from '${url}'`,
+        );
+    }
+    yield* body;
 }
```

Comment:
I mean why use `AsyncIterator` here instead of `Promise`?

But I see, in node it's actually used to yield chunks one by one instead of the whole thing

Reply:
I fail to see how a `Promise` relates to handling data streams

Reply:
Can you provide example code that streams the request body using a `Promise` instead of an iterable?

Reply:
Nevermind, I'm just used to `fetch` returning a promise where `body` is just a value, not a stream